### PR TITLE
add gradient clipping

### PR DESCRIPTION
**Pull request by MorinoseiMorizo**
Added gradient clipping.

### CODE REVIEW

File: sample_data/sample_config.ini [hunk 1]
```diff
@@ -163,6 +163,10 @@ adam_eps=1e-8
 ;            number of sentences in the batch.
 loss_integration_type=sum
 
+; Clip the gradient if the L-2 norm of the gradient is larger than this threshold.
+; Set 0 to disable this.
+gradient_clipping=5
```
Comment (odashi) [inline sample_data/sample_config.ini]: 5.0.

File: src/bin/train.cc [hunk 2]
```diff
@@ -449,6 +449,13 @@ int main(int argc, char * argv[]) {
         config.get<string>("Train.loss_integration_type"));
     logger->info("Created new encoder-decoder model.");
 
+    // Gradient clipping
+    const unsigned gradient_clipping = config.get<unsigned>("Train.gradient_clipping");
```
Comment (odashi) [inline src/bin/train.cc]: const float.

File: src/bin/train.cc [hunk 3]
```diff
@@ -449,6 +449,13 @@ int main(int argc, char * argv[]) {
...
+    if (gradient_clipping != 0) {
```
Comment (odashi) [inline src/bin/train.cc]: > 0.0f

**Comment by MorinoseiMorizo** [comment]
Thank you for the comments.
I fixed it.

**Pull request merged**